agent-control/src/cli/on_host/host_monitoring_gen/infra_config.rs

```diff
@@ -0,0 +1,286 @@
+
+        assert_eq!(result, EXPECTED_AGENT_TYPE_CONFIG);
+    }
+
+    #[test]
+    fn test_generate_infra_config_values() {
+        let custom_attributes = r#"custom_attributes:
+  custom_key: custom_value
+"#;
+        let infra_config = InfraConfig::default()
+            .with_custom_attributes(custom_attributes)
+            .unwrap()
+            .with_region(Region::STAGING)
+            .with_proxy("http://proxy.example.com");
+        let result = infra_config.generate_infra_config_values().unwrap();
+
+        // Parse the YAML content
+        let parsed_values: serde_yaml::Value = serde_yaml::from_str(&result).unwrap();
+
+        if let serde_yaml::Value::Mapping(map) = parsed_values {
+            if let Some(serde_yaml::Value::Mapping(config_agent_map)) =
+                map.get(serde_yaml::Value::String("config_agent".to_string()))
+            {
+                assert_eq!(
+                    config_agent_map.get(serde_yaml::Value::String(
+                        "status_server_enabled".to_string()
+                    )),
+                    Some(&serde_yaml::Value::Bool(true))
+                );
+                assert_eq!(
+                    config_agent_map.get(serde_yaml::Value::String(
+                        "enable_process_metrics".to_string()
+                    )),
+                    Some(&serde_yaml::Value::Bool(true))
+                );
+                assert_eq!(
+                    config_agent_map.get(serde_yaml::Value::String("license_key".to_string())),
+                    Some(&serde_yaml::Value::String(
+                        "{{NEW_RELIC_LICENSE_KEY}}".to_string()
+                    ))
+                );
+                assert_eq!(
+                    config_agent_map
+                        .get(serde_yaml::Value::String("status_server_port".to_string())),
+                    Some(&serde_yaml::Value::Number(serde_yaml::Number::from(18003)))
+                );
+                assert_eq!(
+                    config_agent_map.get(serde_yaml::Value::String("staging".to_string())),
+                    Some(&serde_yaml::Value::Bool(true))
+                );
+                assert_eq!(
+                    config_agent_map.get(serde_yaml::Value::String("proxy".to_string())),
+                    Some(&serde_yaml::Value::String(
+                        "http://proxy.example.com".to_string()
+                    ))
+                );
+                let mut custom_attributes = serde_yaml::Mapping::new();
+                custom_attributes.insert(
+                    serde_yaml::Value::String("custom_key".to_string()),
+                    serde_yaml::Value::String("custom_value".to_string()),
+                );
+                assert_eq!(
+                    config_agent_map
+                        .get(serde_yaml::Value::String("custom_attributes".to_string())),
+                    Some(&serde_yaml::Value::Mapping(custom_attributes))
+                );
+            }
```

Comment:
Did you consider writing a string and simply comparing the `serde_yaml::Values` instead?

```rust
let expected = r"#
config_agent:
  status_server_enabled: true
  # ...
#"
let expected_values: serde_yaml::Value = serde_yaml::from_str(expected).unwrap();
assert_eq!(parsed_values, expected_values);
```